library/VTKExtensions/Rendering/vtkF3DRenderer.cxx

```diff
@@ -1338,6 +1339,22 @@ void vtkF3DRenderer::ShowEdge(bool show)
   }
 }
 
+//----------------------------------------------------------------------------
+void vtkF3DRenderer::SetUseOrthographicProjection(bool use)
+{
+  vtkCamera* camera = GetActiveCamera();
+  if (use)
+  {
+    const double* position = camera->GetPosition();
+    const double* focal = camera->GetFocalPoint();
+    const double angle = camera->GetViewAngle();
+    double distance = std::sqrt(vtkMath::Distance2BetweenPoints(position, focal));
+    double parallelScale = distance * tan(0.5 * vtkMath::RadiansFromDegrees(angle));
+    camera->SetParallelScale(parallelScale);
+  }
```

Comment:
So here we compute the `parallelScale` from the `distance` (to the focal point) when turning parallel projection on?
But we don't do the inverse by repositioning the camera at a `distance` (from the focal point) computed from `parallelScale` when we turn parallel projection off?

It feels like there should be an `else` block going something like
```cpp
double distance = camera->GetParallelScale() / tan(angle/2);
// new_pos = focal + distance * normed(pos - focal)
camera->setPosition(new_pos)
```
@Meakk to confirm?

Reply:
To me it feels like we need a `--camera-parallel-scale` for orthographic mode, just like we have `--camera-view-angle` for perspective mode.  
I don't know where the formula above comes from, but it looks like a heuristic to me.

Reply:
I'm willing to believe the heuristic is good enough, and I think it's needed so the camera doesn't jump back and forth when pressing the toggle key. But it would need to work both ways and I'm not sure it would as is. needs playing around with to find out